core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -66,20 +70,78 @@ public ConstantScoreQueryBuilder boost(float boost) {
         return this;
     }
 
+    /**
+     * @return the boost factor
+     */
+    public float boost() {
+        return this.boost;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("filter");
         filterBuilder.toXContent(builder, params);
+        builder.field("boost", boost);
+        builder.endObject();
+    }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query innerFilter = filterBuilder.toQuery(parseContext);
```

Comment:
I have second thoughts about removing the null checks here.... sorry! let's say a java api user creates a new query without a filter, then calls manually toQuery which makes no sense but it is a public method... what do we do? this case won't be covered by our validation framework. Shall we let it throw NPE or handle it gracefully?